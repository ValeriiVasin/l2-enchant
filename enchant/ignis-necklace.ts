import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const ignisNecklaceEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.BlessedSpiritsVarnish,
  chances: [50, 50, 40, 40, 30, 30, 20, 10, 3, 1],
  items: itemsRange(Item.IgnisNecklace, Item.IgnisNecklace_10),
});
