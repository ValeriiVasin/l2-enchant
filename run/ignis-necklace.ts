import { enchant } from '../helpers/enchant';
import { header } from '../helpers/header';
import { Item } from '../types';

const enchants: Array<[number, Item]> = [
  [1, Item.IgnisNecklace_1],
  [2, Item.IgnisNecklace_2],
  [3, Item.IgnisNecklace_3],
  [4, Item.IgnisNecklace_4],
  [5, Item.IgnisNecklace_5],
  [6, Item.IgnisNecklace_6],
  [7, Item.IgnisNecklace_7],
  [8, Item.IgnisNecklace_8],
];

for (const [lvl, item] of enchants) {
  console.log(header(`Ожерелье Игниса +0 => +${lvl}`));
  console.log(enchant(Item.IgnisNecklace, item, 1000));
}
