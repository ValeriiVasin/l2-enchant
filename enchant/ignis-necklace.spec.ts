import { ignisNecklaceEnchantMap } from './ignis-necklace';

it('enchant map: ignis necklace', () => {
  expect(ignisNecklaceEnchantMap).toMatchSnapshot();
});
