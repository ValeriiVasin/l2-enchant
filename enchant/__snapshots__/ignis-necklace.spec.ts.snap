// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: ignis necklace 1`] = `
Map {
  "Ожерелье Игниса" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Ожерелье Игниса +1",
      },
      Object {
        "chance": 50,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +1" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +1",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Ожерелье Игниса +2",
      },
      Object {
        "chance": 50,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +2" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Ожерелье Игниса +3",
      },
      Object {
        "chance": 60,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +3" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +3",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Ожерелье Игниса +4",
      },
      Object {
        "chance": 60,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +4" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +4",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Ожерелье Игниса +5",
      },
      Object {
        "chance": 70,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +5" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +5",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Ожерелье Игниса +6",
      },
      Object {
        "chance": 70,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +6" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +6",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Ожерелье Игниса +7",
      },
      Object {
        "chance": 80,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +7" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +7",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Ожерелье Игниса +8",
      },
      Object {
        "chance": 90,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +8" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +8",
    "results": Array [
      Object {
        "chance": 3,
        "item": "Ожерелье Игниса +9",
      },
      Object {
        "chance": 97,
        "item": "Ожерелье Игниса",
      },
    ],
  },
  "Ожерелье Игниса +9" => Object {
    "cost": "Благословенный Лак Духов",
    "item": "Ожерелье Игниса +9",
    "results": Array [
      Object {
        "chance": 1,
        "item": "Ожерелье Игниса +10",
      },
      Object {
        "chance": 99,
        "item": "Ожерелье Игниса",
      },
    ],
  },
}
`;
